src/withStyles.jsx

```diff
@@ -37,13 +37,22 @@ export function withStyles(
     pureComponent = false,
   } = {},
 ) {
-  const styleDef = styleFn ? ThemedStyleSheet.create(styleFn) : EMPTY_STYLES_FN;
+  let styleDef;
   const BaseClass = baseClass(pureComponent);
 
   return function withStylesHOC(WrappedComponent) {
     // NOTE: Use a class here so components are ref-able if need be:
     // eslint-disable-next-line react/prefer-stateless-function
     class WithStyles extends BaseClass {
+      constructor(props) {
+        super(props);
+
+        // defer StyleSheet creation to run-time
```

Comment:
can this go in `componentWillMount` instead?

Reply:
I thought that there wasn't a real difference between the two. What would be the advantage of doing it in `componentWillMount`?

Reply:
Conceptually, `componentWillMount` is for side effects, constructors are not. Functionally they're the same.

However, avoiding "needing a constructor at all" is a benefit of using `componentWillMount` here.